src/WebJobs.Script/Description/Workers/ScriptInvocationContext.cs

```diff
@@ -31,5 +32,13 @@ public class ScriptInvocationContext
         public ILogger Logger { get; set; }
 
         public System.Threading.ExecutionContext AsyncExecutionContext { get; set; }
+
+        /// <summary>
+        /// Gets or sets the list of names of <see cref="MemoryMappedFile"/> that were allocated to
+        /// transfer data to the worker process for this invcation.
+        /// These are tracked here so that once the invocation is complete, the resources can be
+        /// freed.
+        /// </summary>
+        public IList<string> SharedMemoryResources { get; set; }
```

Comment:
This list is created and deleted with in the worker channel.  No need to expose this property here

Reply:
The reason for keeping this list here is so that all `SharedMemoryFile`s created for a particular invocation can be passed from `SendInvocationRequest` (where they are created) to `InvocationResponse` (where they are freed.) I don't see a better way to return all these from `ToRpcInvocationRequest` (which is called from `SendInvocationRequest`) to track this list in `GrpcWorkerChannel`.

The alternate I can think of is to have a method in `SharedMemoryManager` to add a mapping between invocation ID and list of `SharedMemoryFile` names (something like `SharedMemoryManager::TrackSharedMemoryFilesForInvocation(string invocationId, IList<string> mapNames)`) which can be called from inside `ToRpcInvocationRequest` where the `SharedMemoryFile`s are created and we already know the invocation ID and also have the `SharedMemoryManager`. Then `GrpcWorkerChannel` can free these from inside `InvocationResponse` by calling another method on `SharedMemoryManager` (something like `SharedMemoryManager::FreeSharedMemoryFilesForInvocation(string invocationId)`).

Thoughts?

Reply:
yes, that is what I am referring to . Define an interface and move the logic that keeps a record of invocationId and memory maps created for that invocation into it. This isolates how a channel manages memory maps into an interface.